File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
       if(n<=3) return n;

       int a=2, b=3;
       int next;
       for(int i=3;i<n;i++){
          next=a+b;
          a=b;
          b=next;
       }
       return next;


    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def fun(self, i, n, dp):
        if i > n:
            return 0

        if i == n:
            return 1

        if dp[i] != -1:
            return dp[i]

        one = self.fun(i + 1, n, dp)
        two = self.fun(i + 2, n, dp)

        dp[i] = one + two
        return dp[i]

    def climbStairs(self, n: int) -> int:
        dp = [-1] * (n + 2)

        return self.fun(0, n, dp)

